power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -1226,10 +1226,10 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         return sc_input;
     }
 
-    template <bool sym> void prepare_y_bus() {
+    template <symmetry_tag sym> void prepare_y_bus() {
         std::vector<YBus<sym>>& y_bus_vec = get_y_bus<sym>();
         // also get the vector of other Y_bus (sym -> asym, or asym -> sym)
-        std::vector<YBus<!sym>>& other_y_bus_vec = get_y_bus<!sym>();
+        std::vector<YBus<other_symmetry_t<sym>>>& other_y_bus_vec = get_y_bus<other_symmetry_t<sym>>();
```

Comment:
we may want to move the declaration of `other_symmetry_t` here

Reply:
That's okay common for such a small struct.